tests/jballerina-unit-test/src/test/resources/test-src/object/ObjectProject/src/init/object-initializer.bal

```diff
@@ -562,3 +562,30 @@ function testConstRefsAsDefaultValue() returns (boolean) {
     [string , int, TYPE, TYPE ] details = s.getDetails();
     return details[0] == "Andrew" && details[1] == 11 && details[2] == "John" && details[3] == 11;
 }
+
+function getData(string n) returns string {
+    return n;
+}
+
+type Student11 object {
+    string name;
+    function __init(string n, function (string) returns string fn = (x) => "John") {
```

Comment:
Shall we add an example, including lambda function as well.

Reply:
+1, Added